Sources/AppKitBackend/AppKitBackend.swift

```diff
@@ -1685,6 +1686,105 @@ public final class AppKitBackend: AppBackend {
+        let sheet = NSCustomSheet(
+            contentRect: NSRect(
+                x: 0,
+                y: 0,
+                width: 400,  // Default width
+                height: 400  // Default height
+            ),
+            styleMask: [.titled, .closable],
+            backing: .buffered,
+            defer: true
+        )
+        sheet.contentView = content
+
+        return sheet
+    }
+
+    public func updateSheet(
+        _ sheet: NSCustomSheet,
+        onDismiss: @escaping () -> Void
+    ) {
+        sheet.onDismiss = onDismiss
+    }
+
+    public func sizeOf(_ sheet: NSCustomSheet) -> SIMD2<Int> {
+        guard let size = sheet.contentView?.frame.size else {
+            return SIMD2(x: 0, y: 0)
+        }
+        return SIMD2(x: Int(size.width), y: Int(size.height))
+    }
+
+    public func showSheet(_ sheet: NSCustomSheet, window: NSCustomWindow?) {
+        guard let window else {
+            print("warning: Cannot show sheet without a parent window")
+            return
+        }
+        // Critical sheets stack. beginSheet only shows a nested sheet
+        // after its parent gets dismissed.
+        window.beginCriticalSheet(sheet)
+    }
+
+    public func dismissSheet(_ sheet: NSCustomSheet, window: NSCustomWindow) {
+        window.endSheet(sheet)
+    }
+
+    public func setPresentationBackground(of sheet: NSCustomSheet, to color: Color) {
+        let backgroundView = NSView()
+        backgroundView.wantsLayer = true
+        backgroundView.layer?.backgroundColor = color.nsColor.cgColor
+
+        if let existingContentView = sheet.contentView {
+            let container = NSView()
+            container.translatesAutoresizingMaskIntoConstraints = false
+
+            container.addSubview(backgroundView)
+            backgroundView.translatesAutoresizingMaskIntoConstraints = false
+            backgroundView.leadingAnchor.constraint(equalTo: container.leadingAnchor).isActive =
+                true
+            backgroundView.topAnchor.constraint(equalTo: container.topAnchor).isActive = true
+            backgroundView.trailingAnchor.constraint(equalTo: container.trailingAnchor).isActive =
+                true
+            backgroundView.bottomAnchor.constraint(equalTo: container.bottomAnchor).isActive = true
+
+            container.addSubview(existingContentView)
+            existingContentView.translatesAutoresizingMaskIntoConstraints = false
+            existingContentView.leadingAnchor.constraint(equalTo: container.leadingAnchor)
+                .isActive = true
+            existingContentView.topAnchor.constraint(equalTo: container.topAnchor).isActive = true
+            existingContentView.trailingAnchor.constraint(equalTo: container.trailingAnchor)
+                .isActive = true
+            existingContentView.bottomAnchor.constraint(equalTo: container.bottomAnchor).isActive =
+                true
+
+            sheet.contentView = container
```

Comment:
I feel like this function should be able to be called twice (e.g. if we start respecting live updates to the preference values of the sheet's content). In its current form, that would mean that we'd repeatedly add levels of NSView containers one on top of another each time the background colour changes (or even worse, every time that it could possibly have changed).

For that reason, I think AppKitBackend should construct this nested background+content container thing as its `contentView` in `createSheet`. You can store a reference to that actual use contentView in a new property on `NSCustomSheet`. That should make this update-background method relatively trivial.